src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -5163,5 +5163,66 @@ public void CompileAndVerifyModuleIncludesAllModules()
                 Assert.Equal("refMod.netmodule", a.ContainingModule.Name);
             });
         }
+
+        [Fact]
+        [WorkItem(37779, "https://github.com/dotnet/roslyn/issues/37779")]
+        public void WarnAsErrorDoesNotEmit()
+        {
+            string source = @"
+class X
+{
+    int _f;
+}";
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, generalDiagnosticOption: ReportDiagnostic.Error);
+            var compilation = CreateCompilation(source, options: options);
+
+            using var output = new MemoryStream();
+            using var pdbStream = new MemoryStream();
+            using var xmlDocumentationStream = new MemoryStream();
+
+            var emitResult = compilation.Emit(output, pdbStream, xmlDocumentationStream, win32Resources: null);
+            Assert.False(emitResult.Success);
+
+            Assert.Equal(0, output.Length);
+            Assert.Equal(0, pdbStream.Length);
+            Assert.Equal(0, xmlDocumentationStream.Length);
```

Comment:
Validation of no binary, pdb and xml doc generated on /warnaserror breaking build from Emit API invocation.